nucypher/blockchain/eth/sol/source/contracts/lib/SignatureVerifier.sol

```diff
@@ -87,4 +87,71 @@ library SignatureVerifier {
         return toAddress(_publicKey) == recover(hash(_message, _algorithm), _signature);
     }
 
+    /**
+    * @notice Hash message according to EIP191 signature specification
+    * @dev It always assumes Keccak256 is used as hashing algorithm
+    * @dev Only supports version 0 and version E (0x45)
+    * @param _message Message to sign
+    * @param _version EIP191 version to use
+    **/
+    function hashEIP191(
+        bytes memory _message,
+        byte _version
+    )
+        internal
+        view
+        returns (bytes32 result)
+    {
+        if(_version == byte(0x00)){  // Version 0: Data with intended validator
+            address validator = address(this);
+            return keccak256(abi.encodePacked(byte(0x19), byte(0x00), validator, _message));
+        } else if (_version == byte(0x45)){  // Version E: personal_sign messages
+            require(_message.length > 0, "Empty message not allowed for version E");
+            // Header for Version E as defined by EIP191. First byte ('E') is also the version
+            bytes25 header = "Ethereum Signed Message:\n";
+
+            // Compute text-encoded length of message
+            uint256 length = _message.length;
```

Comment:
Very small optimisation - put it before require statement and use variable in require